test-app/src/main/java/org/readium/r2/testapp/reader/VisualReaderFragment.kt

```diff
@@ -545,6 +582,56 @@ abstract class VisualReaderFragment : BaseReaderFragment() {
             }
         }
 
+    private fun showFootnote(
+        text: Spanned,
+        point: PointF
+    ) {
+        viewLifecycleOwner.lifecycleScope.launch {
+            viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
```

Comment:
There's no suspending calls here. Did you mean to check for `isAdded`?  I thought the `EventChannel` was handling this already.

Reply:
`repeatOnLifecycle` is a suspending call... This is a way to handle configuration changes.

Reply:
That's true, I guess closing the popup should cancel the job.

Reply:
I've tried some fixes, but I can't reason about lifecycles in `VisualReaderFragment`, so it's presumably broken. Fixing it would require significantly more work (probably a rewrite).